kentik_api_library/kentik_api/api_resources/devices_api.py

```diff
@@ -2,33 +2,69 @@
 from http import HTTPStatus
 from kentik_api.api_resources.base_api import BaseAPI
 from kentik_api.api_calls import devices
-from kentik_api.public.device import Device, AppliedLabels
-from kentik_api.requests_payload import devices_payload
+from kentik_api.public.device import Device, AppliedLabels, Interface
+from kentik_api.requests_payload import devices_payload, interfaces_payload
+from kentik_api.api_connection.api_connector_protocol import APIConnectorProtocol
+
+
+class InterfacesAPI(BaseAPI):
+    """ Exposes Kentik API operations related to interfaces """
+
+    def get_all(self, device_id: int) -> List[Interface]:
+        api_call = devices.get_device_interfaces(device_id)
+        response = self.send(api_call)
+        return interfaces_payload.GetAllResponse.from_json(response.text).to_interfaces()
+
+    def get(self, device_id: int, interface_id: int) -> Interface:
+        api_call = devices.get_device_interface_info(device_id, interface_id)
+        response = self.send(api_call)
+        return interfaces_payload.GetResponse.from_json(response.text).to_interface()
+
+    def create(self, interface: Interface) -> Interface:
+        api_call = devices.create_interface(interface.device_id)
+        payload = interfaces_payload.CreateRequest.from_interface(interface)
+        response = self.send(api_call, payload)
+        return interfaces_payload.CreateResponse.from_json(response.text).to_interface()
+
+    def update(self, interface: Interface) -> Interface:
+        api_call = devices.update_interface(interface.device_id, interface.id)
+        payload = interfaces_payload.UpdateRequest.from_interface(interface)
+        response = self.send(api_call, payload)
+        return interfaces_payload.UpdateResponse.from_json(response.text).to_interface()
+
+    def delete(self, device_id: int, interface_id: int) -> bool:
+        api_call = devices.delete_interface(device_id, interface_id)
+        response = self.send(api_call)
+        return response.http_status_code == HTTPStatus.OK
 
 
 class DevicesAPI(BaseAPI):
     """ Exposes Kentik API operations related to devices """
 
+    def __init__(self, api_connector: APIConnectorProtocol) -> None:
+        super(DevicesAPI, self).__init__(api_connector)
+        self._interfaces = InterfacesAPI(api_connector)
```

Comment:
Nit: I'd prefer `InterfaceAPI` class definition below `DevicesAPI` definition, because it is a field of DevicesAPI. My personal preference is to put type definitions below its usage. Could you put `InterfaceAPI` class definition below `DevicesAPI`?

Reply:
there is same situation in CustomDimensions, lets change this in next PR ok?

Reply:
update: cant do that:
E   NameError: name 'InterfacesAPI' is not defined

Reply:
Well, this is unfortunate.